### ISSUE TITLE
Request: Add `doas` backend

### ISSUE THREAD

**Issue by jboynyc** [digint/btrbk#444]
To make it easier to deploy btrbk on servers that for security purposes do not have `sudo`, it would be great to add a backend option for `doas`.

**Comment by digint** [comment]
I suppose all commands would have to prefix `doas -n`, analogous to `sudo -n` ?
https://github.com/digint/btrbk/blob/master/btrbk#L321-L330

I should probably also add a patch to ssh_filter_btrbk.

For even more security, I suggest having a look at btrfs-progs-btrbk, which eliminates the need of sudo/doas by setting linux capabilities (setcap):
https://github.com/digint/btrfs-progs-btrbk

**Comment by jboynyc** [comment]
Yes, `doas -n` is non-interactive mode, works the same as `sudo`.

I saw the `btrfs-progs` option, but unfortunately it isn't feasible for my use case.

**Comment by digint** [comment]
Implemented `backend btrfs-progs-doas` in [backend-doas branch](https://github.com/digint/btrbk/tree/backend-doas).

Please give it a try:
```
cd /tmp/
wget https://raw.githubusercontent.com/digint/btrbk/backend-doas/btrbk
chmod +x btrbk
# add "backend btrfs-progs-doas" in /etc/btrbk.conf
./btrbk run -n
```

**Comment by jboynyc** [comment]
Thanks so much, I will give it a try.

**Comment by jboynyc** [comment]
I've now been able to try it, and everything worked as hoped! Thanks again.

**Comment by digint** [comment]
thanks, merged to master

**Comment by digint** [comment]
included in btrbk-0.32.1
